packages/fx-core/templates/core/v3Migration/js.ts.app.yml

```diff
@@ -28,19 +35,82 @@ deploy:
       workingDirectory: tabs
       args: install
   - uses: cli/runNpmCommand # Run npm command
+    {{#if activePlugins.fx-resource-aad-app-for-teams}}
     env:
       REACT_APP_CLIENT_ID: $\{{AAD_APP_CLIENT_ID}}
       REACT_APP_START_LOGIN_PAGE_URL: $\{{ {{~placeholderMappings.[state.fx-resource-frontend-hosting.endpoint]~}} }}/auth-start.html
+      {{#if activePlugins.fx-resource-function}}
+      REACT_APP_FUNC_ENDPOINT: $\{{ {{~placeholderMappings.[state.fx-resource-function.functionEndpoint]~}} }}
+      REACT_APP_FUNC_NAME: {{defaultFunctionName}}
+      {{/if}}
+    {{/if}}
     with:
       workingDirectory: tabs
-      args: run build
+      args: run build --if-present
   - uses: azureStorage/deploy # Deploy bits to Azure Storage Static Website
     with:
       workingDirectory: tabs
       distributionPath: build # Deploy base folder. This folder includes manifest files for AAD app and Teams app that should be ignored using the ignoreFile.
       ignoreFile:  # Can be changed to any ignore file location, leave blank will ignore nothing
       resourceId: $\{{ {{~placeholderMappings.[state.fx-resource-frontend-hosting.storageResourceId]~}} }} # The resource id of the cloud resource to be deployed to
 {{/if}}
+{{#if activePlugins.fx-resource-bot}}
+  {{#if isFunctionBot}}
+  - uses: cli/runNpmCommand
+    with:
+      args: install
+    {{#if isTypescript}}
+  - uses: cli/runNpmCommand
+    with:
+      args: run build --if-present
+    {{/if}}
+  - uses: azureFunctions/deploy
+    with:
+      distributionPath: ./ # Deploy base folder
+      ignoreFile: .funcignore # Can be changed to any ignore file location, leave blank will ignore nothing
+      # the resource id of the cloud resource to be deployed to
+      resourceId: $\{{ {{~placeholderMappings.[state.fx-resource-bot.functionAppResourceId]~}} }}
+  {{else}}
+  - uses: cli/runNpmCommand # Run npm command
+    with:
+      workingDirectory: bot
+      args: install
+    {{#if isTypescript}}
+  - uses: cli/runNpmCommand # Run npm command
+    with:
+      workingDirectory: bot
+      args: run build --if-present
+    {{/if}}
+  - uses: azureAppService/deploy # Deploy bits to Azure Storage Static Website
+    with:
+      workingDirectory: bot
+      distributionPath: . # Deploy base folder. 
+      ignoreFile: # Can be changed to any ignore file location, leave blank will ignore nothing
```

Comment:
Deploy to Azure App Service with a nodeJS project requires at least one ignore file that ignores the .deployment folder

Reply:
thanks, let me track this separately